unliftio/src/UnliftIO/Async.hs

```diff
@@ -713,38 +813,40 @@ runFlat f0 = E.uninterruptibleMask $ \restore -> do
   --
   let autoRetry action =
         action `E.catch`
-        \E.BlockedIndefinitelyOnSTM -> autoRetry action
+        \case E.BlockedIndefinitelyOnSTM -> autoRetry action
```

Comment:
Why is this change needed?

Reply:
In the (unlikely) event there are more constructors added to the `BlockedIndefinitelyOnSTM` type, this change will throw a warning on compilation time

Reply:
Thanks for the clarification, can you add a comment to the codebase to avoid refactoring it away in the future? Also, isn't there a GHC warning that can be turned on that will do this?

Reply:
> Also, isn't there a GHC warning that can be turned on that will do this?

You are right, there is ` -fwarn-incomplete-uni-patterns` for this kind of situations as well

Reply:
I think I'd rather use the warning in the cabal/package.yaml file then. Using LambdaCase here feels like a hack. And the warning is more explicit, and will help the rest of the codebase too.